libsolidity/parsing/Parser.cpp

```diff
@@ -823,9 +823,15 @@ ASTPointer<ParameterList> Parser::parseParameterList(
 	VarDeclParserOptions options(_options);
 	options.allowEmptyName = true;
 	expectToken(Token::LParen);
+
 	if (!_allowEmpty || m_scanner->currentToken() != Token::RParen)
 	{
 		parameters.push_back(parseVariableDeclaration(options));
+		
+		if(m_scanner->currentToken() == Token::Comma && m_scanner->peekNextToken() == Token::RParen){
```

Comment:
It looks like this check only happens at the beginning of the parameter list, instead of the end.

Could you please add some tests to verify that?

Also, you accidentally added a binary file to this PR. Please remove it and rebase.

Reply:
Yes, will do.

Reply:
Right, fixed that. Will add boost tests and re-request